### PR TITLE
Adding support for singularity containers

### PR DESCRIPTION
**Pull request by alecgunny**

### CODE REVIEW

File: omicron/cli/process.py [hunk 1]
```diff
@@ -1058,14 +1067,16 @@ def main(args=None):
     if not args.skip_rm:
         rmjob = condor.OmicronProcessJob(
             args.universe, str(condir / "post-process-rm.sh"),
-            subdir=condir, logdir=logdir, tag='post-processing-rm', **condorcmds)
+            subdir=condir, logdir=logdir, tag='post-processing-rm',
+            singularity_image=args.singulatiy_image, **condorcmds)
```
Comment (EthanMarx) [inline omicron/cli/process.py]: Small typeo here: `singulatiy_image` --> `singularity_image`

**Review by EthanMarx — CHANGES_REQUESTED** [review]
Looks great. One minor typeo fix and should be good. 

I am new to apptainer so want to clarify the picture here: 

We will be launching the pyomicron process inside the `/ml4gw/bbhnet` `apptainer` image. This will create and submit a condor dag which contains omicron jobs that run inside this image, and point to the omicron executable we have installed inside the image.

Is this setup compatible with using the LDG condor scheduler/ accessing all the LDG nodes?

File: omicron/cli/process.py (hunk 1 above)
Comment (alecgunny) [reply omicron/cli/process.py]: good catch

**Comment by alecgunny** [comment]
Yeah that's exactly it. As far as I understand, it should be so long as you bind the `/etc/condor` directory into the container used to launch the jobs so that it has access to the appropriate condor configs, e.g.
```console
apptainer exec --bind /etc/condor:/etc/condor /cvmfs/singularity.opensciencegrid.org/ml4gw/bbhnet:main ...
```

**Review by EthanMarx — APPROVED** [review]

**Review by EthanMarx — APPROVED** [review]

**Comment by EthanMarx** [comment]
Think I need write access for my review to count

**Pull request merged**